### Q1 What version of Pandas did you install?


In [10]:
import pandas as pd

In [11]:
pd.__version__

'3.0.2'

### Resposta: versão 3.0.2

### Importando os dados da atividade 

In [12]:
car_df = pd.read_csv("car_fuel_efficiency.csv")

# Conferindo  se foi importado puxando as 5 primeiras linhas 

car_df.head()

,engine_displacement,num_cylinders,horsepower,vehicle_weight,acceleration,model_year,origin,fuel_type,drivetrain,num_doors,fuel_efficiency_mpg
0,170,3.0,159.0,3413.433759,17.7,2003,Europe,Gasoline,All-wheel drive,0.0,13.231729
1,130,5.0,97.0,3149.664934,17.8,2007,USA,Gasoline,Front-wheel drive,0.0,13.688217
2,170,NaN,78.0,3079.038997,15.1,2018,Europe,Gasoline,Front-wheel drive,0.0,14.246341
3,220,4.0,NaN,2542.392402,20.2,2009,USA,Diesel,All-wheel drive,2.0,16.912736
4,210,1.0,140.0,3460.870990,14.4,2009,Europe,Gasoline,All-wheel drive,2.0,12.488369


### Q2 How many records are in the dataset?

In [15]:
# Usando describe

car_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9704 entries, 0 to 9703
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   engine_displacement  9704 non-null   int64  
 1   num_cylinders        9222 non-null   float64
 2   horsepower           8996 non-null   float64
 3   vehicle_weight       9704 non-null   float64
 4   acceleration         8774 non-null   float64
 5   model_year           9704 non-null   int64  
 6   origin               9704 non-null   str    
 7   fuel_type            9704 non-null   str    
 8   drivetrain           9704 non-null   str    
 9   num_doors            9202 non-null   float64
 10  fuel_efficiency_mpg  9704 non-null   float64
dtypes: float64(6), int64(2), str(3)
memory usage: 834.1 KB


#### answer: 9704 records

### q3 How many fuel types are presented in the dataset?

In [17]:
fuel_count = car_df['Fuel'].nunique()

KeyError: 'Fuel'